python/src/helpers/count_matrix.py

```diff
@@ -108,10 +86,15 @@ def download_object(self, key, etag):
                 Fileobj=f,
             )
 
+            self.last_fetch = datetime.datetime.now(timezone.utc)
             f.seek(0)
 
+        if was_enabled:
```

Comment:
Why do this twice?

Reply:
No, `was_enabled` is only checked in line 78. If it was already enabled, we disable and re-enable. If it was never enabled, we don't disable or reenable.

Reply:
Yeah that's what I meant, in the second one you set it to false again instead of re-enabling it

Reply:
oh, good catch! fixed :)